File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        low = 0
        high = len(height)-1
        res = 0
        while (low < high):
            vol = (high-low)*min(height[low], height[high])
            res = max(res, vol)
            if height[low] < height[high]:
                low += 1
            else:
                high -= 1
        
        return res

File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int low = 0;
        int high = height.size()-1;
        int res = 0;
        while (low < high) {
            int vol = min(height[low], height[high])*(high-low);
            res = max(res, vol);
            if (height[low] < height[high]) {
                low++;
            } else {
                high--;
            }
        }
        return res;
    }
};